### PR TITLE
review comments: break-word for long file names

### LINKED ISSUE #16248: Incorrect wrapping on PR comments with long paths
- Gitea version (or commit ref): 1.14.3
- Git version: 2.25.1
- Operating system: Ubuntu Linux 20.04.2
- Built Gitea myself, running from systemd (my own service script) on an AWS x86_64 box
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes: https://try.gitea.io/parnic-sks/long-paths-test-2/pulls/1
  - [ ] No

## Description

Comments on files with long paths will not wrap the file name header on a PR if there's no easily-available word boundary to wrap on. This causes the "Outdated" badge and/or "Show Resolved"/"Hide Resolved" links to either overlap other UI elements or be completely obscured, rendering them un-clickable. Note: this is not a contrived example, we ran into this on our project.

Example of a long path with available word boundaries: https://try.gitea.io/parnic-sks/long-paths-test/pulls/1
Example of a long path that doesn't currently wrap: https://try.gitea.io/parnic-sks/long-paths-test-2/pulls/1

## Screenshots

Working (valid word boundaries):
image

Not working (no valid word boundaries):
image

## Potential fix

Setting `word-break: break-word;` CSS on one of the `segment` bits used for PR comment filename headers is one possible way to fix the issue:
image

### PR DESCRIPTION
**Pull request by noerw**
Use break-word for long file names in review comment headers.

#### after

Fixes #16248

### CODE REVIEW

File: web_src/less/_base.less [hunk 1]
```diff
@@ -2071,3 +2071,7 @@ table th[data-sortt-desc] {
   margin-top: -.5em;
   margin-bottom: -.5em;
 }
+
+.break-word {
```
Comment (zeripath) [inline web_src/less/_base.less]: There is a .word-break in helpers.less

File: templates/repo/issue/view_content/comments.tmpl [hunk 2]
```diff
@@ -462,7 +462,7 @@
 				{{ range $filename, $lines := .Review.CodeComments}}
 					{{range $line, $comms := $lines}}
 							<div class="ui segments">
-								<div class="ui segment py-3 df ac sb">
+								<div class="ui segment py-3 df ac sb break-word">
```
Comment (zeripath) [inline templates/repo/issue/view_content/comments.tmpl]:
```suggestion
								<div class="ui segment py-3 df ac sb word-break">
```

File: web_src/less/_base.less [hunk 3]
```diff
@@ -2071,3 +2071,7 @@ table th[data-sortt-desc] {
...
+  word-break: break-word;
+}
```
Comment (zeripath) [inline web_src/less/_base.less]:
```suggestion
```

**Review by zeripath — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Comment by parnic-sks** [comment]
This fix doesn't seem to have taken. The class is now applied at https://try.gitea.io/parnic-sks/long-paths-test-2/pulls/1, but the text still doesn't break.

**Comment by 6543** [comment]
haha it's only fixed for firefox :1st_place_medal:

**Pull request merged**